sites/texas_tribune.py

```diff
@@ -76,14 +79,72 @@ def bulk_fetch(start_date: date, end_date: date) -> List[Dict[str, Any]]:
     API_URL = f"https://{DOMAIN}/api/v2/articles"
     DATE_FORMAT = "%Y-%m-%dT%H:%M:%S"
     # texas tribune publishes 5-10 articles per day
-    params = {"start_date": start_date.strftime(DATE_FORMAT), "end_date": end_date.strftime(DATE_FORMAT), "limit": 100}
+    params = {
+        "start_date": start_date.strftime(DATE_FORMAT),
+        "end_date": end_date.strftime(DATE_FORMAT),
+        "limit": BULK_FETCH_LIMIT,
+    }
     res = safe_get(API_URL, params=params, scrape_config=SCRAPE_CONFIG)
     json_res = res.json()
 
     metadata = [parse_metadata(article) for article in json_res["results"]]
     return metadata
 
 
+# Added to accommodate SS
+def get_article_text(metadata: Dict[str, Any]) -> str:
+    """
+    Get text representation of any article.
+    """
+    return metadata["headline"] + ". " + metadata["summary"]
+
+
+# Added to accommodate SS
+def batch_fetch_by_external_id(batch_external_ids: List[str]) -> Dict[str, Any]:
+    """
+    Helper for bulk_fetch_by_external_id. Sends a request for a batch of IDs.
+    """
+    # For instance: https://www.texastribune.org/api/v2/articles/?id=40916&id=40930 returns both of the articles with the corresponding IDs
+    query = "&".join([f"id={i}" for i in batch_external_ids])
+    url = f"https://{DOMAIN}/api/v2/articles/?{query}"
+    params = {"limit": BULK_FETCH_LIMIT}
+
+    # Request
+    res = safe_get(url, params=params, scrape_config=SCRAPE_CONFIG)
+    error_msg = validate_response(res, [validate_status_code])
+
+    if error_msg is not None:
```

Comment:
🔧 I think this can this be simplified to `if error_msg`. If `error_msg` is anything truth-y, it proceeds as desired. Given that the return type of `validate_response` is `Optional[str]` that means it should only either be `None` or a `str`, so yeah go ahead and simplify to just checking `if error_msg`.

Reply:
Hmm, I kind of prefer to keep `is not None` there. It makes the if-condition clearer at the expense of three words. It also accounts for the edge case of the falsy empty string `""` being assigned to `error_msg`. What do you think?

Reply:
Sure, not my preference but makes sense 👍🏿